Nodejs/Product/Nodejs/NpmUI/NpmWorker.cs

```
@@ -106,74 +106,135 @@ private void Run()
-                            "name": "express",
-                            "scope": "unscoped",
-                            "version": "4.15.2",
-                            "description": "Fast, unopinionated, minimalist web framework",
-                            "keywords": [ "express", "framework", "sinatra", "web", "rest", "restful", "router", "app", "api" ],
-                            "date": "2017-03-06T13:42:44.853Z",
-                            "links": {
-                              "npm": "https://www.npmjs.com/package/express",
-                              "homepage": "http://expressjs.com/",
-                              "repository": "https://github.com/expressjs/express",
-                              "bugs": "https://github.com/expressjs/express/issues"
-                            },
-                            "author": {
-                              "name": "TJ Holowaychuk",
-                              "email": "tj@vision-media.ca"
-                            },
-                            "publisher": {
-                              "username": "dougwilson",
-                              "email": "doug@somethingdoug.com"
-                            },
-                            "maintainers": [
-                              {
-                                "username": "dougwilson",
-                                "email": "doug@somethingdoug.com"
-                              }
-                            ]
-                          },
-                          "score": {
-                            "final": 0.9549640105248649,
-                            "detail": {
-                              "quality": 0.9427473299991661,
-                              "popularity": 0.9496544159654299,
-                              "maintenance": 0.9707450455348992
+                            switch (jsonReader.TokenType)
+                            {
+                                case JsonToken.StartObject:
```

Do we want to throw if this happens a second time?

---

No, don't really want to build a state machine for an assert

---

Could you not just assert that the stream is done after the object is read?

---

I don't really care if there's more stuff send by the NPM api after the package I care about... And considering they can (and have) changed the response without warning, i prefer to be as lenient as possible.

---

We appear to throw if we see anything other than an object. 😛

---

True, since that's the thing we care about... so silently failing would be worse.